src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -162,16 +167,126 @@ public Integer getPayeeNameFieldSize() {
         String parameterValue = getParameterService().getParameterValueAsString(CUKFSConstants.ParameterNamespaces.CONCUR, 
                 CUKFSParameterKeyConstants.ALL_COMPONENTS, parameterName);
-        if (LOG.isDebugEnabled()) { 
-            LOG.debug("getConcurParamterValue, parameterName: " + parameterName + " paramterValue: " + parameterValue);
-        }
         return parameterValue;
     }
 
     public void setPayeeNameFieldSize(Integer payeeNameFieldSize) {
         this.payeeNameFieldSize = payeeNameFieldSize;
     }
     
+    @Override
+    public PdpFeedFileBaseEntry createPdpFileBaseEntryThatDoesNotContainNonReimbursableSections(PdpFeedFileBaseEntry pdpFeedFileBaseEntry, 
+            ConcurStandardAccountingExtractBatchReportData reportData) {
+        LOG.debug("Entering removeNonReimbursableSectionsFromPdpFeedFileBaseEntry");
+        PdpFeedFileBaseEntry newBaseEntry = new PdpFeedFileBaseEntry();
+        newBaseEntry.setHeader(copyHeaderEntry(pdpFeedFileBaseEntry.getHeader()));
+        newBaseEntry.setVersion(pdpFeedFileBaseEntry.getVersion());
+        addGroupEntriesToNewBaseEntry(newBaseEntry, pdpFeedFileBaseEntry);
+        newBaseEntry.setTrailer(buildPdpFeedTrailerEntry(newBaseEntry, reportData));
+        return newBaseEntry;
+    }
+
+    private void addGroupEntriesToNewBaseEntry(PdpFeedFileBaseEntry newBaseEntry, PdpFeedFileBaseEntry originalBaseEntry) {
+        for (PdpFeedGroupEntry originalGroupEntry : originalBaseEntry.getGroup()) {
+            PdpFeedGroupEntry newGroupEntry = copyGroupEntry(originalGroupEntry);
+            addDetailEntriesToNewGroupEntry(newGroupEntry, originalGroupEntry);
+            if (CollectionUtils.isNotEmpty(newGroupEntry.getDetail())) {
+                newBaseEntry.getGroup().add(newGroupEntry);
+            } else {
+                LOG.debug("addGroupEntriesToNewBaseEntry, not adding group for" + newGroupEntry.getPayeeName());
+            }
+        }
+    }
+    
+    private void addDetailEntriesToNewGroupEntry(PdpFeedGroupEntry newGroupEntry, PdpFeedGroupEntry originalGroupEntry) {
+        for (PdpFeedDetailEntry originalDetailEntry : originalGroupEntry.getDetail()) {
+            PdpFeedDetailEntry newDetailEntry = copyDetailEntry(originalDetailEntry);
+            KualiDecimal originalDetailTransactionTotal = KualiDecimal.ZERO;
+            for (PdpFeedAccountingEntry originalAccountingEntry : originalDetailEntry.getAccounting()) {
+                originalDetailTransactionTotal = originalDetailTransactionTotal.add(originalAccountingEntry.getAmount());
+                addOnlyPositiveAccountingEntriesForPDPProcessing(newDetailEntry, originalAccountingEntry);
+            }
+            addNewPaymentDetailToGroupIfTotalIsPositive(newGroupEntry, newDetailEntry, originalDetailTransactionTotal);
+        }
+    }
+
+    private void addOnlyPositiveAccountingEntriesForPDPProcessing(PdpFeedDetailEntry newDetailEntry,
```

Comment:
This method is only adding up to one accounting entry at a time, even though the name implies that multiple entries could be added. It might make sense to rename this method to reflect that. Sorry for not noticing that in the suggestions for the previous round of review.